src/sqlancer/postgres/gen/PostgresWindowFunctionGenerator.java

```diff
@@ -0,0 +1,96 @@
+package sqlancer.postgres.gen;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import sqlancer.Randomly;
+import sqlancer.postgres.PostgresGlobalState;
+import sqlancer.postgres.PostgresSchema.PostgresDataType;
+import sqlancer.postgres.ast.PostgresExpression;
+import sqlancer.postgres.ast.PostgresOrderByTerm;
+import sqlancer.postgres.ast.PostgresWindowFunction;
+import sqlancer.postgres.ast.PostgresWindowFunction.WindowFrame;
+import sqlancer.postgres.ast.PostgresWindowFunction.WindowSpecification;
+
+public class PostgresWindowFunctionGenerator {
+    
+    private static final List<String> WINDOW_FUNCTIONS = Arrays.asList(
+        "row_number", "rank", "dense_rank", "percent_rank",
+        "cume_dist", "ntile", "lag", "lead", "first_value",
+        "last_value", "nth_value"
+    );
+
+    public static PostgresWindowFunction generateWindowFunction(PostgresGlobalState globalState,
+            List<PostgresExpression> availableExpr) {
+        
+        String functionName = Randomly.fromList(WINDOW_FUNCTIONS);
+        List<PostgresExpression> arguments = new ArrayList<>();
+        
+        // Generate function arguments based on function name
```

Comment:
Also here and below, I think it would improve readability to extract common blocks into methods instead of using comments.